lib/repl.js

```diff
@@ -94,6 +94,7 @@ function REPLServer(prompt, stream, eval_, useGlobal, ignoreUndefined) {
     ignoreUndefined = options.ignoreUndefined;
     prompt = options.prompt;
     dom = options.domain;
+    recoverableCheck = options.recoverable || isRecoverableError;
```

Comment:
This would be better as a ternary based on whether `options.recoverable` is a function or not.